### PR TITLE
Add additional RPC API methods

### PR DESCRIPTION
**Pull request by biancabuzea200**
https://app.clickup.com/t/2645698/DEV-8136

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-8136 document RPC API methods](https://app.clickup.com/t/2645698/DEV-8136)

File: docs/standards/rpc-api.md [hunk 1]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+   <tr>
+    <td><a href="#up_addTransactionRelayer">up_addTransactionRelayer</a></td>
+    <td>LUKSO specific</td>
+  </tr>
+   <tr>
+    <td><a href="#up_import">up_import</a></td>
+    <td>LUKSO specific</td>
+  </tr>
+   <tr>
+    <td><a href="#up_generateLsp23Address">up_generateLsp23Address</a></td>
```
Comment (asciiman) [inline docs/standards/rpc-api.md:55]: This method is missing in the description view below.

Comment (biancabuzea200) [reply docs/standards/rpc-api.md:55]: yes, that one is not yet documented. I got some info from Lukas, but it still needs more work. I wanted to add what I have so far and build incrementally

Comment (asciiman) [reply docs/standards/rpc-api.md:55]: Okay cool. Makes sense.

The rest looks great!

**Review by asciiman — APPROVED** [review]
LGTM!

File: docs/standards/rpc-api.md [hunk 2]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+### personal_sign
+
+The personal_sign endpoint is enabled to allow for backward compatibility. However, its use is not recommended.
+Some libraries such as Ethers.js end up using personal_sign under the hood. That is why, for compatibility reasons, personal_sign is left enabled. Note however that personal_sign ultimately acts as a proxy, redirecting the call to the [eth_sign](#eth_sign) method.
+
+## Standard
+
+### eth_sendTransaction {#eth_sendTransaction}
+
+Creates new message call transaction or a contract creation, if the data field contains code, and signs it using the account specified in `from`. This method requires that the user has granted permission to interact with their account first, so make sure to call eth_requestAccounts first.
```
Comment (JeneaVranceanu) [inline docs/standards/rpc-api.md:83]: > or a contract creation, if the data field contains code

It's not only the data field that highlights a transaction as "contract creation" transaction, e.g. `to` address should be empty when it's a deployment of a contract. 

I'd leave out the "contract creation" part from this documentation.

Also, I think that's important to mention, that the transaction will not be signed by the UP itself but by the controller address that has enough permissions to act on said UP. 

And I'm not sure what you meant by making sure the user granted permission and making "sure to call eth_requestAccounts first.". Even though `eth_requestAccounts` could be informative it doesn't influence granted permissions. Happy to discuss 🙂

File: docs/standards/rpc-api.md [hunk 3]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+
+### eth_requestAccounts {#eth_requestAccounts}
+
+This method is specified by [EIP-1102](https://eips.ethereum.org/EIPS/eip-1102).
+It requests that the user provides an Ethereum address to be identified by.
+
+Returns
+`string[]` - if the user accepts the request, it returns an array of a single, hexadecimal Ethereum address string, or
+4001 - if the user rejects the request
```
[the remark this answers is not in the document]
Comment (JeneaVranceanu) [reply docs/standards/rpc-api.md:92]: https://eips.ethereum.org/EIPS/eip-1102#:~:text=This%20method%20returns%20a%20Promise%20that%20is%20resolved%20with%20an%20Array%20of%20accounts%20or%20is%20rejected%20with%20an%20Error

[the remark this answers is not in the document]
Comment (JeneaVranceanu) [reply docs/standards/rpc-api.md:92]: Hmm, I guess the major difference between `eth_getAccounts` and `eth_requestAccounts` is that the latter expects the wallet to ask a user which accounts he/she wants to make visible to the DApp that makes this request.
While the `eth_getAccounts` just expected to return immediately with all accounts available. Seems, that `eth_getAccounts` was, as many of those RPCs initially available, used by the nodes that had no UI or were remotely located away from the users.
Makes me wonder if we even want to support `eth_getAccounts` 🤔 

Also, I found `eth_accounts` and `personal_listAccounts` 🥲 
Glad we don't support all of them. 1 is enough.

File: docs/standards/rpc-api.md [hunk 4]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+
+Returns
+`null` or `error` - the method will return null if successful or throw an error otherwise
+
+### eth_getAccounts {#eth_getAccounts}
+
+This method returns all of the addresses that returns a list of accounts the node controls.
+
+Returns
+`string[][]` - a successful request returns an array of hexadecimal Ethereum address strings
```
Comment (JeneaVranceanu) [inline docs/standards/rpc-api.md:109]: I think it should be just `string[]`.

File: docs/standards/rpc-api.md [hunk 5]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+It allows Dapps to request that a wallet switches its active chain (connection).
+
+The method requires that a target chain ID is provided
+
+Returns
+`null` or `error` - the method will return null if successful or throw an error otherwise
+
+### eth_getAccounts {#eth_getAccounts}
+
+This method returns all of the addresses that returns a list of accounts the node controls.
```
Comment (JeneaVranceanu) [inline docs/standards/rpc-api.md:106]: It feels like this document is primarily focused on RPCs we support in our apps (extension, mobile or anything else user-facing that should have RPCs support).

Maybe this RPC will be or is already supported by validator/archive/other nodes for some purpose, I'm not sure, but I think this is not the right doc to highlight that. Please, correct me if I'm wrong. I'm open for a discussion.

Comment (JeneaVranceanu) [reply docs/standards/rpc-api.md:106]: Oh, yes, this doc is about what LUKSO extension supports. Judging by the intro sentence.

File: docs/standards/rpc-api.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (biancabuzea200) [reply docs/standards/rpc-api.md:92]: so what should we have here, @JeneaVranceanu ? Maybe something like in the EIP1102:
"Calling this method may trigger a user interface that allows the user to approve or reject account access for a given dapp. This method returns an array of accounts or is rejected with an error code 4001 if accounts are not available." ?
Let me know if you have a better suggestion, please

[the remark this answers is not in the document]
Comment (JeneaVranceanu) [reply docs/standards/rpc-api.md]: I suggest the following:
```
### eth_requestAccounts {#eth_requestAccounts}

This method is specified by [EIP-1102](https://eips.ethereum.org/EIPS/eip-1102).
Calling this method may trigger a user interface that allows the user to approve or reject account access for a given DApp.

Returns
`string[]` - an array of accounts or throws an error with code `4001` if the request was rejected by the user.
```

I left out `if accounts are not available` from the last sentence as I do not think we are ever going to have this case.

File: docs/standards/rpc-api.md [hunk 6]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
```
Comment (JeneaVranceanu) [inline docs/standards/rpc-api.md:106]:
```suggestion
Similar to the `eth_requestAccounts` this method returns all of the addresses that are controlled by the application.
```

File: docs/standards/rpc-api.md [hunk 7]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+
+This method returns a [EIP-191](https://eips.ethereum.org/EIPS/eip-191) signature over the data provided to the call.
+It requests that the user provides an Ethereum address that should sign the transaction as well as the data (encoded bytes) that are to be executed.
+
+Returns
+`string` - on a successful call the method returns a signature, a string representing hex encoded bytes or an error with code 4001 - if the user rejects the requets
+
+### personal_sign
+
+The personal_sign endpoint is enabled to allow for backward compatibility. However, its use is not recommended.
```
Comment (JeneaVranceanu) [inline docs/standards/rpc-api.md:76]: Similar to how Hugo wrapped eth_sign into `` let's wrap all other pieces of code or RPC names into `` as well, like `personal_sign` here.

Those that are wrapped in HTML tags or are links `[text](link)` we can leave as is.

File: docs/standards/rpc-api.md [hunk 8]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+
+This method implements [EIP-3326](https://eips.ethereum.org/EIPS/eip-3326).
+It allows Dapps to request that a wallet switches its active chain (connection).
+
+The method requires that a target chain ID is provided
+
+Returns
+`null` or `error` - the method will return null if successful or throw an error otherwise
+
+### eth_getAccounts {#eth_getAccounts}
```
[the remark this answers is not in the document]
Comment (biancabuzea200) [reply docs/standards/rpc-api.md:104]: ok, will wait, no worries

File: docs/standards/rpc-api.md [hunk 9]
```diff
@@ -10,14 +10,109 @@ sidebar_position: 10
...
+This method returns all of the addresses that returns a list of accounts the node controls.
+
+Returns
+`string[]` - a successful request returns an array of hexadecimal Ethereum address strings
+
+## LUKSO Specific
+
+### up_addTransactionRelayer {#up_addTransactionRelayer}
+
+<p id="up_addTransactionRelayer">Add a custom relayer.</p>
```
[the remark this answers is not in the document]
Comment (biancabuzea200) [reply docs/standards/rpc-api.md:115]: removed

File: docs/standards/rpc-api.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (JeneaVranceanu) [reply docs/standards/rpc-api.md:104]: Okay, so `eth_getAccounts` appears to be one more typo. I haven't virtually anything related to `eth_getAccounts` in Ethereum organisation on GitHub:
Screenshot 2023-09-07 at 13 53 14


But have found a lot regarding the `eth_accounts`:
Screenshot 2023-09-07 at 13 53 25

So we should use `eth_accounts` instead.
Keep the description, just change the name, please.

UP extension correctly uses `eth_accounts`.
We need to update the mobile and RPCs tracking list.

[the remark this answers is not in the document]
Comment (biancabuzea200) [reply docs/standards/rpc-api.md:104]: we should update here as well https://www.notion.so/lukso/d735b2388ee64849b9e920eaf5a50833?v=2211bb0db3d745638e6a349e734cda02

**Review by Hugoo — APPROVED** [review]

**Comment by Hugoo** [comment]
nice pr

**Pull request merged**